test/basic_functionality.jl

```diff
@@ -5,12 +5,13 @@ m,n,k,s = 1000,1000,5,1000*1000;
 # matrix to encode
 X_real, Y_real = randn(m,k), randn(k,n);
 A = X_real*Y_real;
-losses = fill(quadratic(),n);
+losses = convert(Array{Loss,1}, fill(quadratic(),n));
```

Comment:
Would defining GLRM as a parametric type work?
```
type GLRM{L<:Loss, R<:Regularizer}
    A::Array{Float64,2}          # The data table transformed into a coded array 
    losses::Array{L,1}        # array of loss functions
    rx::R              # The regularization to be applied to each row of Xᵀ (column of X)
    ry::Array{R,1}     # Array of regularizers to be applied to each column of Y
    k::Int                     # Desired rank 
    observed_features::ObsArray  # for each example, an array telling which features were observed
    observed_examples::ObsArray  # for each feature, an array telling in which examples the feature was observed  
    X::Array{Float64,2}          # Representation of data in low-rank space. A ≈ X'Y
    Y::Array{Float64,2}          # Representation of features in low-rank space. A ≈ X'Y
end
```
See [this post](https://groups.google.com/forum/#!searchin/julia-users/array$20subtypes$20abstract/julia-users/zQHFQ46al5s/nhtfx0IRZM8J).